snmp/__init__.py

```diff
@@ -0,0 +1,339 @@
+#!/usr/bin/env python3
+# vim: set encoding=utf-8 tabstop=4 softtabstop=4 shiftwidth=4 expandtab
+#########################################################################
+#  Copyright 2012-2014 Marcus Popp                         marcus@popp.mx
+#  Copyright 2019-2020 Michael Wenzel               wenzel_michael@web.de
+#########################################################################
+#  This file is part of SmartHomeNG.    https://github.com/smarthomeNG//
+#
+#  SmartHomeNG is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  SmartHomeNG is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with SmartHomeNG. If not, see <http://www.gnu.org/licenses/>.
+#########################################################################
+
+from lib.module import Modules
+from lib.model.smartplugin import *
+from lib.item import Items
+
+import threading
+import logging
+import time
+import struct
+from puresnmp import get
+
+class Snmp(SmartPlugin):
+    """
+    Main class of the Plugin. Does all plugin specific stuff and provides
+    the update functions for the items
+    """
+
+    ALLOW_MULTIINSTANCE = True
+    PLUGIN_VERSION = '1.6.0'
+
+    _flip = {0: '1', False: '1', 1: '0', True: '0', '0': True, '1': False}
+
+    _supported = {
+        'value': 'Value',           #Wert wird erwartet
+        'string': 'String'}         #String wird erwartet; Rückmeldewert in String oder Bytearray wie bspw b'TS-251'
+    
+    def __init__(self, sh, *args, **kwargs ):
+        """
+        Initalizes the plugin. The parameters describe for this method are pulled from the entry in plugin.conf.
+        """
+
+        from bin.smarthome import VERSION
+        if '.'.join(VERSION.split('.', 2)[:2]) <= '1.5':
+            self.logger = logging.getLogger(__name__)
+
+        if self.logger.isEnabledFor(logging.DEBUG):
+            self.logger.debug("init {}".format(__name__))
+        self._sh = self.get_sh()
```

Comment:
do you really need that line?

Reply:
which line your are talking about?

Reply:
self.logger.debug(...) does the trick of logging only if debug logging is enabled. self.logger.isEnabledFor(logging.DEBUG) has the same effect so it is redundant.

Reply:
OK, I will remove that and do a new PR.
Are there more thing to do?